democracy/views/hearing.py

```diff
@@ -99,8 +100,25 @@ def get_serializer(self, *args, **kwargs):
         kwargs['context'] = self.get_serializer_context()
         return serializer_class(*args, **kwargs)
 
+    @staticmethod
+    def check_preview_code(id, code):
+        import base64
+        import hashlib
+        check = base64.urlsafe_b64encode(
+            hashlib.sha256(
+                "{}{}".format(id, settings.SECRET_KEY).encode('utf-8')
+            ).digest()
+        ).decode('utf-8').replace('=', '')
+        return check == code
+
     def get_queryset(self):
```

Comment:
Would there maybe be a neater way to implement this if you overrode `get_object()` instead of `get_queryset()`? After all, `?preview=` should only work for the detail view, not lists etc.